llumnix/backends/vllm/llm_engine.py

```diff
@@ -199,7 +213,13 @@ def add_request(self, request_id: str, server_info: ServerInfo, *args, **kwargs)
                                         seq_group.metrics.arrival_time, seq_group.lora_request, seq_group.multi_modal_data)
         self.scheduler.scheduler_lock.release()
 
-    def _put_request_outputs_to_server(self, request_outputs, server_infos: List[ServerInfo]) -> None:
+    def _start_put_queue_loop(self):
```

Comment:
Can we add a batch parameter to enable sending multiple tokens?

Reply:
Also, notice that if batching mode is enabled, you'll need to add a timeout to the queue get call, otherwise early tokens could be blocked for too long if later tokens didn't come

Reply:
Do you mean that batched multiple steps or batched multiple tokens? Although batched put will reduce overhead, however it will make the token latency become higher. Considering that put queue operation can be overlaped with engine step(need to be tested further), this batching mode might be not necessary.